pkg/vm/stackitem/json.go

```diff
@@ -260,72 +261,110 @@ func (d *decoder) decodeMap() (*Map, error) {
 
 // ToJSONWithTypes serializes any stackitem to JSON in a lossless way.
 func ToJSONWithTypes(item Item) ([]byte, error) {
-	result, err := toJSONWithTypes(item, make(map[Item]bool, typicalNumOfItems))
-	if err != nil {
-		return nil, err
-	}
-	return json.Marshal(result)
+	return toJSONWithTypes(nil, item, make(map[Item]sliceNoPointer, typicalNumOfItems))
 }
 
-func toJSONWithTypes(item Item, seen map[Item]bool) (interface{}, error) {
+func toJSONWithTypes(data []byte, item Item, seen map[Item]sliceNoPointer) ([]byte, error) {
+	if item == nil {
+		return nil, fmt.Errorf("%w: nil", ErrUnserializable)
+	}
 	if len(seen) > MaxJSONDepth {
-		return "", ErrTooDeep
+		return nil, ErrTooDeep
+	}
+	if old, ok := seen[item]; ok {
+		if old.end == 0 {
+			// Compound item marshaling which has not yet finished.
+			return nil, ErrRecursive
+		}
+		if len(data)+old.end-old.start > MaxSize {
+			return nil, errTooBigSize
+		}
+		return append(data, data[old.start:old.end]...), nil
+	}
+
+	if _, ok := item.(Null); ok {
+		if len(data)+14 > MaxSize {
+			return nil, errTooBigSize
+		}
+		data = append(data, `{"type":"Any"}`...)
+		return data, nil
+	}
+
+	start := len(data)
+
+	prefix := `{"type":"` + item.Type().String() + `","value":`
```

Comment:
This creates a string, why don't we append it to `data` right away? We know it won't be large.

Reply:
Mostly for consistence. This an error exactly here can be expected to be rare, though. Fixed.